File: VocaDbModel/DataContracts/WebhookContract.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using VocaDb.Model.Domain;

namespace VocaDb.Model.DataContracts
{
	public sealed record WebhookContract
	{
		public string Url { get; init; } = string.Empty;

		[JsonConverter(typeof(StringEnumConverter))]
		public WebhookEvents WebhookEvents { get; init; } = WebhookEvents.User;

		public WebhookContract() { }

		public WebhookContract(Webhook webhook)
		{
			Url = webhook.Url;
			WebhookEvents = webhook.WebhookEvents;
		}
	}
}

File: VocaDbWeb/Scripts/DataContracts/WebhookContract.ts
export enum WebhookEvents {
	Default = 0,
	User = 1 << 0,
	EntryReport = 1 << 1,
	DuplicateAccount = 1 << 2,
}

export interface WebhookContract {
	url: string;
	webhookEvents: string;
}
